mewarpx/mewarpx/mwxrun.py

```diff
@@ -103,23 +104,41 @@ def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
         return self.dt
 
 
-    def init_run(self):
+    def init_run(self, restart=None, checkpoint_dir="diags", checkpoint_prefix="chkpoint", additional_steps=None):
         if self.initialized:
             raise RuntimeError(
                 "Attempted to initialize the mwxrun class multiple times.")
-
-        self.simulation.initialize_inputs()
-        self.simulation.initialize_warpx()
-
-        self.me = _libwarpx.libwarpx.warpx_getMyProc()
-        self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
-
-        # A level is needed for many things like level number. For now I'm
-        # statically setting the default level here. I'm not sure of pitfalls
-        # or how to handle it more generally yet.
-        self.lev = _libwarpx.libwarpx.warpx_finestLevel()
-
-        self.initialized = True
+        try:
+            if restart != False:
+                init_restart_util.run_restart(
+                    checkpoint_dir, checkpoint_prefix,
+                    restart is not None, additional_steps
+                )
```

Comment:
Seeing as the `run_restart` function returns a boolean of whether the simulation will be restarted we can use it later for the appropriate logging statement
```suggestion
                restart = init_restart_util.run_restart(
                    checkpoint_dir, checkpoint_prefix,
                    restart is not None, additional_steps
                )
```